packages/x-date-pickers/src/internals/hooks/useField/useField.utils.ts

```diff
@@ -180,6 +205,26 @@ export const adjustInvalidDateSectionValue = <TDate, TSection extends FieldSecti
       return utils.formatByString(newDate, section.formatValue);
     }
 
+    case 'weekDay': {
+      let newDate: TDate;
+      if (shouldSetAbsolute) {
+        if (delta > 0 || isEnd) {
+          newDate = utils.startOfWeek(today);
+        } else {
+          newDate = utils.endOfWeek(today);
+        }
+
+        return utils.formatByString(newDate, section.formatValue);
+      }
+
+      const formattedDaysInWeek = getDaysInWeekStr(utils, section.formatValue);
+      const currentDayInWeek = formattedDaysInWeek.indexOf(section.value);
+      const newDayInWeek =
```

Comment:
There might be some check missing here, or is this an expected behavior? 🤔 

Or it could also be some check missing in `applyWeekDayChange` function.

https://user-images.githubusercontent.com/4941090/215624285-cc70a227-6f2d-4b50-8617-aee83fbbc1e5.mov

Reply:
Nice catch !
I had a bug in the digit editing, when using a letter week day section :+1:

Reply:
The bug is fixed! 🙌 
I can no longer spot or anticipate any further issues. 💯